packages/server/src/database/migrations/20200212131402-normalize-users.js

```diff
@@ -0,0 +1,127 @@
+        },
+        createdAt: {
+          allowNull: false,
+          type: Sequelize.DATE
+        },
+        updatedAt: {
+          allowNull: false,
+          type: Sequelize.DATE
+        }
+      }
+    )
+
+    await queryInterface.createTable('UserAddresses',
+      {
+        id: {
+          allowNull: false,
+          autoIncrement: true,
+          primaryKey: true,
+          type: Sequelize.INTEGER
+        },
+        address: {
+          type: Sequelize.STRING,
+          allowNull: false,
+          unique: true,
+        },
+        userId: {
+          type: Sequelize.INTEGER,
+          references: {
+            model: 'Users',
+            key: 'id',
+            as: 'userId',
+          },
+        },
+        createdAt: {
+          allowNull: false,
+          type: Sequelize.DATE
+        },
+        updatedAt: {
+          allowNull: false,
+          type: Sequelize.DATE
+        }
+      }
+    )
+
+    console.log('')
+    const subscriptions = (await queryInterface.sequelize.query(`SELECT * FROM "Subscriptions" ORDER BY id`))[0]
+
+    for (const { email, address, createdAt, updatedAt } of subscriptions) {
+      const parsedEmail = email.toLowerCase()
+      const existingUser = (await queryInterface.sequelize.query(`SELECT id FROM "Users" WHERE email='${parsedEmail}'`))[0]
+
+      if (existingUser.length === 0) {
+        logger.success(`Inserting new user with email '${parsedEmail}'`)
+        await queryInterface.sequelize.query(`INSERT INTO "Users" ("email", "createdAt", "updatedAt") VALUES ('${parsedEmail}', '${createdAt.toUTCString()}', '${updatedAt.toUTCString()}')`)
+      }
+      else {
+        const { id: userId } = existingUser[0]
+        logger.warn(`Skipping email '${parsedEmail}', user with ID '${userId}' already has that email address`)
+      }
+
+      const { id: userId } = (await queryInterface.sequelize.query(`SELECT id FROM "Users" WHERE email='${parsedEmail}'`))[0][0]
+      if (!address) logger.warn(`Skipping address '${address}' for user with ID '${userId}'`)
+      else {
+        const parsedAddress = isAddress(address) ? toChecksumAddress(address) : address
+        const existingAddress = (await queryInterface.sequelize.query(`SELECT * FROM "UserAddresses" WHERE address='${parsedAddress}'`))[0]
+
+        if (existingAddress.length === 0) {
+          logger.success(`Inserting new address '${parsedAddress}' for user with ID '${userId}' with email '${parsedEmail}'`)
+          await queryInterface.sequelize.query(`INSERT INTO "UserAddresses" ("userId", "address", "createdAt", "updatedAt") VALUES ('${userId}', '${parsedAddress}', '${createdAt.toUTCString()}', '${updatedAt.toUTCString()}')`)
+        } else {
+          const { id: addressId, userId: addressUserId } = existingAddress[0]
+          logger.warn(`Skipping address '${parsedAddress}' for user with ID '${userId}', another user with ID '${addressUserId}' already has that address with ID '${addressId}'`)
```

Comment:
I think there might be another case here where it's because the address has already been added to the `UserAddresses` table—should we check against `existingAddress.some(address => address === parsedAddress)`?

Reply:
The problem is that if the address was already added, it means it has an email associated. I think we must support multiple addresses per email, but not multiple emails per address, i.e. 1-N email-address relation. That's exactly why [this case](https://github.com/aragonone/court-backend/pull/73#discussion_r398966540) is possible. If we find the same address linked to multiple emails, we will have the first email linked to the address, but the rest will be left without one.